### ISSUE TITLE
Failed to install latest version of onnx-weekly nightly job.

### ISSUE THREAD

**Issue by fatcat-z** [onnx/onnx#4053]
# Bug Report

### Describe the bug
In some case, when we run the command "pip install --index-url https://test.pypi.org/simple/ onnx-weekly" to install latest version of onnx-weekly nightly build, it failed to install those latest ones but only an old version 1.10.1.dev20210830 will be installed. Some logs are shown as below:

```
:~/Desktop/code/tensorflow-onnx$ pip install --index-url https://test.pypi.org/simple/ onnx-weekly
Looking in indexes: https://test.pypi.org/simple/
WARNING: Keyring is skipped due to an exception: Failed to create the collection: Prompt dismissed..
Collecting onnx-weekly
Downloading https://test-files.pythonhosted.org/packages/19/56/84b1eb141b6129ffeac8c0c3a76e4170771d003479857ef1d855fa3b09b9/onnx_weekly-1.11.0.dev20220228-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.8 MB)
|████████████████████████████████| 12.8 MB 1.0 MB/s
Downloading https://test-files.pythonhosted.org/packages/3d/13/392cc9c308014001ce5d74e914bdb8d7a917b23f2edb29ac56d4b28c0ed6/onnx_weekly-1.11.0.dev20220221-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.8 MB)
|████████████████████████████████| 12.8 MB 1.1 MB/s
Downloading https://test-files.pythonhosted.org/packages/30/af/2a359cc38da90572c000ff10ac4e939d3423e7d1215300afcfe07044acc5/onnx-weekly-1.11.0.dev20220214.tar.gz (9.9 MB)
|████████████████████████████████| 9.9 MB 87 kB/s
ERROR: Command errored out with exit status 1:
command: /home/anaconda3/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_u6v0gfw/onnx-weekly_9e3e31dab4a743679efa5a3fc060864b/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_u6v0gfw/onnx-weekly_9e3e31dab4a743679efa5a3fc060864b/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-oheeqf44
cwd: /tmp/pip-install-_u6v0gfw/onnx-weekly_9e3e31dab4a743679efa5a3fc060864b/
Complete output (31 lines):
fatal: not a git repository (or any of the parent directories): .git
Traceback (most recent call last):
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/installer.py", line 75, in fetch_build_egg
subprocess.check_call(cmd)
File "/home/anaconda3/lib/python3.8/subprocess.py", line 364, in check_call
raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/home/anaconda3/bin/python', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmpm3gv4xk1', '--quiet', 'pytest-runner']' died with <Signals.SIGSEGV: 11>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "<string>", line 1, in <module>
File "/tmp/pip-install-_u6v0gfw/onnx-weekly_9e3e31dab4a743679efa5a3fc060864b/setup.py", line 336, in <module>
setuptools.setup(
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py", line 152, in setup
_install_setup_requires(attrs)
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
dist.fetch_build_eggs(dist.setup_requires)
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/dist.py", line 686, in fetch_build_eggs
resolved_dists = pkg_resources.working_set.resolve(
File "/home/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 766, in resolve
dist = best[req.key] = env.best_match(
File "/home/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1051, in best_match
return self.obtain(req, installer)
File "/home/anaconda3/lib/python3.8/site-packages/pkg_resources/__init__.py", line 1063, in obtain
return installer(requirement)
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/dist.py", line 745, in fetch_build_egg
return fetch_build_egg(self, req)
File "/home/anaconda3/lib/python3.8/site-packages/setuptools/installer.py", line 77, in fetch_build_egg
raise DistutilsError(str(e)) from e
distutils.errors.DistutilsError: Command '['/home/anaconda3/bin/python', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmpm3gv4xk1', '--quiet', 'pytest-runner']' died with <Signals.SIGSEGV: 11>.
...

```

### System information
- OS Platform and Distribution (*e.g. Linux Ubuntu 16.04*):  Linux Ubuntu 20.04
- ONNX version (*e.g. 1.7*):  nightly build
- Python version:  3.8

### Reproduction instructions
- Execute below command in some environments:
```
pip install --index-url https://test.pypi.org/simple/ onnx-weekly
```

### Expected behavior
The installed onnx-weekly version should be the first one nnx_weekly-1.11.0.dev20220228-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.

### Notes

Any additional information

List all of dependencies version:

+ pip freeze --all
absl-py==0.15.0
astunparse==1.6.3
attrs==21.4.0
cachetools==4.2.4
certifi==2021.10.8
charset-normalizer==2.0.12
clang==5.0
coverage==6.3.2
flatbuffers==1.12
gast==0.4.0
google-auth==1.35.0
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
graphviz==0.19.1
grpcio==1.44.0
h5py==3.1.0
idna==3.3
importlib-metadata==4.11.1
iniconfig==1.1.1
keras==2.6.0
Keras-Preprocessing==1.1.2
Markdown==3.3.6
numpy==1.19.5
oauthlib==3.2.0
onnx-weekly==1.10.1.dev20210830
onnxruntime==1.9.0
onnxruntime-extensions==0.3.1
opt-einsum==3.3.0
packaging==21.3
pandas==1.4.1
parameterized==0.8.1
Pillow==9.0.1
pip==21.2.4
pluggy==1.0.0
protobuf==3.19.4
py==1.11.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyparsing==3.0.7
pytest==7.0.1
pytest-cov==3.0.0
pytest-runner==6.0.0
python-dateutil==2.8.2
pytz==2021.3
PyYAML==6.0
requests==2.27.1
requests-oauthlib==1.3.1
rsa==4.8
setuptools==58.0.4
six==1.15.0
tensorboard==2.6.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
tensorflow==2.6.2
tensorflow-estimator==2.6.0
tensorflow-hub==0.12.0
tensorflow-text==2.6.0
tensorflowjs==3.13.0
termcolor==1.1.0
tf2onnx==1.10.0
tomli==2.0.1
typing-extensions==3.7.4.3
urllib3==1.26.8
Werkzeug==2.0.3
wheel==0.37.1
wrapt==1.12.1
zipp==3.7.0

**Comment by jcwchen** [comment]
HI @fatcat-z.
It seems like an issue for pip that sometimes it cannot fetch the latest package.
Could you please try these commands to avoid download older package:
```
pip install -U pip
pip install --no-cache-dir --upgrade --index-url https://test.pypi.org/simple/ onnx-weekly
```

**Comment by fatcat-z** [comment]
> HI @fatcat-z. It seems like an issue for pip that sometimes it cannot fetch the latest package. Could you please try these commands to avoid download older package:
> 
> ```
> pip install -U pip
> pip install --no-cache-dir --upgrade --index-url https://test.pypi.org/simple/ onnx-weekly
> ```

Sure, I could try this.

But according to above logs, looks like it was trying every newer versions one by one but some of them could not be downloaded and some of them caused an error during installation. Do you know how those errors were thrown out?

**Comment by fatcat-z** [comment]
Hi @jcwchen ,

Tried the command you suggested but the latest version of onnx-weekly still could not be installed if the tensorflow version is 2.6.0 or 2.6.2.

On the same machine and same environment, if I installed the tensorflow 2.8.0, then the latest nightly build of onnx-weekly could be installed successfully. So this issue should be relative to the version of tensorflow, or its dependencies.

**Comment by jcwchen** [comment]
Thank you for the clarification and investigation! Since the latest successful version is 1.10.1.dev20210830, my best guess is it happened due to this PR merged on 9/1: https://github.com/onnx/onnx/pull/3684. Previously, ONNX does not require minimum-supported protobuf version, but at certain point ONNX (when the protobuf version that release CI uses has been bumped to 3.16.0) cannot work with protobuf<3.12 anymore. That's why I added a minimum-supported protobuf as 3.12.2. 

In short: perhaps for some reasons TensorFlow 2.6.x requires lower protobuf version? If it does not, could you please try to upgrade your protobuf version higher than 3.12.2 before install onnx-weekly in the CI environment? Thanks!

**Comment by fatcat-z** [comment]
> Thank you for the clarification and investigation! Since the latest successful version is 1.10.1.dev20210830, my best guess is it happened due to this PR merged on 9/1: #3684. Previously, ONNX does not require minimum-supported protobuf version, but at certain point ONNX (when the protobuf version that release CI uses has been bumped to 3.16.0) cannot work with protobuf<3.12 anymore. That's why I added a minimum-supported protobuf as 3.12.2.
> 
> In short: perhaps for some reasons TensorFlow 2.6.x requires lower protobuf version? If it does not, could you please try to upgrade your protobuf version higher than 3.12.2 before installing onnx-weekly in the CI environment? Thanks!

For the above 2 tests I tried, both of them has the same version 3.19.4 of protobuf, which satisfies the minimum supported version 3.12.2.

**Comment by jcwchen** [comment]
Thank you for the quick experiment. I have another guess is because of introducing wget on 8/30: https://github.com/onnx/onnx/pull/3663. wget does not have a wheel on TestPyPI so without installing it manually first installing ONNX wheel on TestPyPI will fail. Although now latest ONNX does not need wget anymore, it needs NumPy >= 1.21.5 after 2/18: https://github.com/onnx/onnx/pull/4001. That's why it cannot install any onnx-weekly after 8/30... Could you please try two experiments:
1. Use NumPy >= 1.21.5
2. pip install wget before installing onnx-weekly package

If my theory is correct, using 1. should work. Thank you for the experiment! Ideally pip should complain why it cannot install newer versions but somehow it did not.

**Comment by jcwchen** [comment]
The PR has just been merged. After this weekend, the new onnx-weekly package should loosen the requirement for NumPy and then tensorflow-onnx should be able to use it with older TensorFlow versions.
